auth0/resource_auth0_guardian.go

```diff
@@ -118,24 +128,46 @@ func updateGuardian(d *schema.ResourceData, m interface{}) (err error) {
 			err = api.Guardian.MultiFactor.UpdatePolicy(&management.MultiFactorPolicies{p})
 		}
 	}
-	//TODO: Extend for other MFA types
+	if err := updatePhoneFactor(d, api); err != nil {
+		return err
+	}
+	if err := updateEmailFactor(d, api); err != nil {
+		return err
+	}
+	return readGuardian(d, m)
+}
+
+func updatePhoneFactor(d *schema.ResourceData, api *management.Management) error {
 	ok, err := factorShouldBeUpdated(d, "phone")
 	if err != nil {
 		return err
 	}
 	if ok {
-		api.Guardian.MultiFactor.Phone.Enable(true)
+		if err := api.Guardian.MultiFactor.Phone.Enable(true); err != nil {
+			return err
+		}
 		if err := configurePhone(d, api); err != nil {
 			return err
 		}
 	} else {
-		api.Guardian.MultiFactor.Phone.Enable(false)
+		if err := api.Guardian.MultiFactor.Phone.Enable(false); err != nil {
+			return err
+		}
 	}
-	return readGuardian(d, m)
+	return nil
 }
 
-func configurePhone(d *schema.ResourceData, api *management.Management) (err error) {
+func updateEmailFactor(d *schema.ResourceData, api *management.Management) error {
+	if changed := d.HasChange("email"); changed {
+		enabled := d.Get("email").(bool)
+		if err := api.Guardian.MultiFactor.Email.Enable(enabled); err != nil {
+			return err
+		}
```

Comment:
We can return here directly and remove 1 extra indentation level. 
```suggestion
		return api.Guardian.MultiFactor.Email.Enable(enabled)
```

Reply:
Definitely easier on the eyes with reducing the nesting, thanks for pointing this out! Had to change this slightly, as phone needs to be enabled before it's configured. Also added `api.Guardian.MultiFactor.Phone.Enable(false)` for the negative case, as this means it's being removed from state and we need to toggle it off.